generator/src/lib.rs

```diff
@@ -2144,6 +2198,23 @@ pub fn write_source_code(path: &Path) {
     let source_code = quote! {
         use std::fmt;
         use std::os::raw::*;
+        /// Iterates through the pointer chain. Includes the item that is passed into the function.
+        /// Stops at the last `BaseOutStructure` that has a null `p_next` field.
+        pub unsafe fn ptr_chain_iter<T>(
```

Comment:
Is there a good reason for this to be part of the public API?

Reply:
It is used in the test api. I am not sure, is it even possible to share `pub(crate)` code in the *external* tests module? I could move the tests into `lib.rs`, and it should work with `use super::vk`.

Reply:
That sounds like a better solution to me. You could also have an internal tests module, rather than cramming everything into lib.rs.

We should be trying to minimize the amount of code the generator emits (because it's inconvenient to manipulate) and the amount of public API that doesn't have a specific strong motivation (for ease of maintenance).

Reply:
Agreed, a lot of the trait / macro definitions could be defined outside of the generator.